packages/hasher/src/hashOfFiles.ts

```diff
@@ -1,18 +1,28 @@
-import crypto from "crypto";
 import path from "path";
+import { getPackageDeps } from "@rushstack/package-deps-hash";
 import globby from "globby";
-import fs from "fs-extra";
-
 import { hashStrings } from "./helpers";
+import { fastFindUp } from "./workspaces/fastFindUp";
 
-const newline = /\r\n|\r|\n/g;
-const LF = "\n";
+const repoHashes = new Map<string, { [key: string]: string }>();
 
 // We have to force the types because globby types are wrong
 export async function generateHashOfFiles(
   packageRoot: string,
   globs: string[]
 ): Promise<string> {
+  let gitRoot = await fastFindUp(".git", packageRoot);
+
+  if (!gitRoot) {
+    throw new Error("backfill require this to be inside a git repo");
+  }
+
+  gitRoot = path.dirname(gitRoot);
+
+  if (!repoHashes.has(gitRoot!)) {
+    repoHashes.set(gitRoot, getPackageDeps(gitRoot).files);
```

Comment:
Does this include untracked, modified and stages files, or just the committed files?

Reply:
This will use git hash-object which is done for any folks under the git repo whether tracked or not